modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/messaging/test/IndividualSegmentsCheckerTest.java

```diff
@@ -0,0 +1,293 @@
+									"totalElements", 1
+								).put(
+									"totalPages", 1
+								)
+							).put(
+								"total", 0
+							).toString()
+						).build()));
+
+			ReflectionTestUtil.invoke(
+				_individualSegmentsChecker, "checkIndividualSegments",
+				new Class<?>[0]);
+
+			List<SegmentsEntry> segmentsEntryList =
+				_segmentsEntryLocalService.getSegmentsEntries(
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+			Assert.assertEquals(
+				segmentsEntryList.toString(), 1, segmentsEntryList.size());
+
+			SegmentsEntry segmentsEntry = segmentsEntryList.get(0);
+
+			Assert.assertEquals(
+				"Test segment",
+				segmentsEntry.getName(LocaleUtil.getSiteDefault()));
+
+			List<SegmentsEntryRel> segmentsEntryRelList =
+				_segmentsEntryRelLocalService.getSegmentsEntryRels(
+					segmentsEntry.getSegmentsEntryId());
+
+			Assert.assertEquals(
+				segmentsEntryRelList.toString(), 1,
+				segmentsEntryRelList.size());
+
+			SegmentsEntryRel segmentsEntryRel = segmentsEntryRelList.get(0);
+
+			Assert.assertEquals(
+				_user.getUserId(), segmentsEntryRel.getClassPK());
+		}
+	}
+
+	private Object _getInstance(String className, Bundle bundle)
+		throws Exception {
+
+		Class<?> clazz = bundle.loadClass(className);
+
+		Constructor<?> constructor = clazz.getConstructor();
+
+		return constructor.newInstance();
+	}
+
+	private void _setIndividualSegmentsCheckerTest() throws Exception {
+		Bundle bundle1 = FrameworkUtil.getBundle(
+			IndividualSegmentsCheckerTest.class);
+
+		BundleContext bundleContext = bundle1.getBundleContext();
+
+		Bundle bundle2 = null;
+
+		for (Bundle curBundle : bundleContext.getBundles()) {
+			if (Objects.equals(
+					curBundle.getSymbolicName(),
+					"com.liferay.segments.asah.connector")) {
+
+				bundle2 = curBundle;
+
+				break;
+			}
+		}
+
+		Assert.assertNotNull(
+			"Unable to find com.liferay.segments.asah.connector bundle",
+			bundle2);
+
+		_individualSegmentsChecker = _getInstance(
+			"com.liferay.segments.asah.connector.internal.messaging." +
+				"IndividualSegmentsChecker",
+			bundle2);
+
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_analyticsSettingsManager",
+			_analyticsSettingsManager);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_asahSegmentsEntryCache",
+			_getInstance(
+				"com.liferay.segments.asah.connector.internal.cache." +
+					"AsahSegmentsEntryCache",
+				bundle2));
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_companyLocalService",
+			_companyLocalService);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_moduleServiceLifecycle",
+			_moduleServiceLifecycle);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_portal", _portal);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_segmentsEntryLocalService",
+			_segmentsEntryLocalService);
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsChecker, "_segmentsEntryRelLocalService",
```

Comment:
Why do you need to override the  _segmentsEntryRelLocalService use the one in the _individualSegmentsChecker?